principal-component-with-sklearn-in-python/index.md

```diff
@@ -0,0 +1,151 @@
+##  Implementing Principal Component Analysis (*PCA*) algorithm with sklearn in python
+### Introduction to *PCA*
+In Machine learning, it is common to come across data sets with hundreds or even thousands of features. Implementing models on such datasets can be challenging in terms of computational cost. Also, models build on high dimensional datasets are prone to the problem of the course of dimensionality. To minimize this problem, we use a technique known as **Dimensionality Reduction**.
+*Dimensionality Reduction* involves transforming features of a high-dimensional space dataset to a low-dimensional space. Some of the techniques used in the *dimensionality reduction* include *PCA, Linear Discriminant Analysis(LDA), Kernal PCA, Conical Correlation Analysis (CCA),* etc. Of all these techniques, *PCA* is the most used technique in the *dimension reduction* problem.
+
+The *PCA* algorithm reduces the dimension of a dataset by projecting a d-dimensional space dataset onto k- dimensional subspace were $(k<d)$. This algorithm is used in the problems such as *feature extraction, stock market prediction, Gene analysis*, e.t.c.
+
+PCA involve the following steps:
+- Standardizing the data
+- Determining the Eigenvalues and the Eigenvectors from the covariance or correlation matrix.
+- Sorte the eigenvalues in descending order.
+-  Choose the first k eigenvectors corresponding to the k largest eigenvalues where k is the dimension of the new features space, and it's such that $(k<d)$.
+-  Constructing a projection matrix M from the selected k eigenvectors.
+-  Transform the original dataset X via M to obtain the new k-dimensional feature subspace Y. 
+For more knowledge on the mathematical concept behind the PCA algorithm, I recommend you read this [article](https://www.section.io/engineering-education/image-compression-using-pca/).
+
+This article will teach how to implement the PCA algorithm in python using the sklearn library. In this implementation, we will work with a structured dataset that we can download [here]().
+
+### Prerequisites:
+- Python installed on your computer
+- Python programming skills
+- A dataset
+  
+### Implementing *PCA*
+### Step 1: Data preprocessing
+The dataset we are applying PCA on in this session consists of features of different types of wine, with the target variable consisting of the customer segment to which a wine belongs. We can download this dataset [here]()
+
```

Comment:
Before pre-processing, explain about the dataset. Use `head()` and display the first 5 rows. Explain what we are going to do with the dataset